presto-hive/src/test/java/com/facebook/presto/hive/metastore/InMemoryHiveMetastore.java

```diff
@@ -88,8 +91,40 @@ public synchronized void createDatabase(Database database)
         checkArgument(directory.mkdirs(), "Could not create database directory");
 
         if (databases.putIfAbsent(database.getName(), database) != null) {
-            throw new IllegalArgumentException("Database " + database.getName() + " already exists");
+            throw new SchemaAlreadyExistsException(database.getName());
+        }
+    }
+
+    @Override
+    public synchronized void dropDatabase(String databaseName)
+    {
+        if (!databases.containsKey(databaseName)) {
+            throw new SchemaNotFoundException(databaseName);
+        }
+        if (!getAllTables(databaseName).orElse(ImmutableList.of()).isEmpty()) {
+            throw new IllegalArgumentException("Schema not empty: " + databaseName);
+        }
+        databases.remove(databaseName);
+    }
+
+    @Override
+    public synchronized void alterDatabase(String databaseName, Database newDatabase)
+    {
+        if (databaseName.equals(newDatabase.getName())) {
+            if (databases.replace(databaseName, newDatabase) == null) {
+                throw new SchemaNotFoundException(databaseName);
+            }
+            return;
+        }
+
+        Database database = databases.get(databaseName);
+        if (database == null) {
+            throw new SchemaNotFoundException(databaseName);
+        }
+        if (databases.putIfAbsent(newDatabase.getName(), database) != null) {
+            throw new SchemaAlreadyExistsException(newDatabase.getName());
         }
+        databases.remove(databaseName);
```

Comment:
You need to rewrite all of the tables, partitions and permissions.  I'd just fail on this for now.